src/ApiService/ApiService/Log.cs

```diff
@@ -1,418 +1,435 @@
+    /// <param name="state"></param>
+    /// <param name="exception"></param>
+    /// <param name="formatter"></param>
+    /// <exception cref="ArgumentNullException"></exception>
+    void ILogger.Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter) {
+        if (formatter == null) {
+            throw new ArgumentNullException(nameof(formatter));
         }
-    }
-
-    private static void LogTags(Guid correlationId, IReadOnlyDictionary<string, string> tags) {
-        var ts = DictToString(tags);
-        if (!string.IsNullOrEmpty(ts)) {
-            System.Console.WriteLine($"[{correlationId}] Tags:{ts}");
+        foreach (var config in this.telemetryConfig) {
+            if (state is RequestTelemetry request && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Request))) {
+                PopulateTags(request);
+                config.TelemetryClient.TrackRequest(request);
+            } else if (state is PageViewTelemetry pageView && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.PageView))) {
+                PopulateTags(pageView);
+                config.TelemetryClient.TrackPageView(pageView);
+            } else if (state is AvailabilityTelemetry availability && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Availability))) {
+                PopulateTags(availability);
+                config.TelemetryClient.TrackAvailability(availability);
+            } else if (state is DependencyTelemetry dependency && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Dependency))) {
+                PopulateTags(dependency);
+                config.TelemetryClient.TrackDependency(dependency);
+            } else if (state is MetricTelemetry metric && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Metric))) {
+                PopulateTags(metric);
+                config.TelemetryClient.TrackMetric(metric);
+            } else if (state is EventTelemetry evt && (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Event))) {
+                PopulateTags(evt);
+                config.TelemetryClient.TrackEvent(evt);
+            } else {
+                if ((this as ILogger).IsEnabled(logLevel)) {
+                    if (exception is null) {
+                        if (config.EnabledTelemetry is null || config.EnabledTelemetry.Contains(Telemetry.Trace)) {
+                            TraceTelemetry traceTelemetry = new TraceTelemetry(
+                                formatter(state, exception),
+                                OneFuzzLogger.GetSeverityLevel(logLevel));
+
+                            traceTelemetry.Context.Operation.Id = Activity.RootId;
+                            traceTelemetry.Context.Operation.ParentId = Activity.SpanId.ToString();
```

Comment:
I'm wondering if this should be: 
```suggestion
                            traceTelemetry.Context.Operation.Id = Activity.Id;
                            traceTelemetry.Context.Operation.ParentId = Activity.Parent?.Id;
```
?

Reply:
Span ID is correct

![image](https://github.com/microsoft/onefuzz/assets/7307755/fb1d347e-0050-44aa-b453-a6e76d0e8982)

Root ID <- is most parent ID, which also should be correct since the most parent ID